src/server/http/setup_connection.js

```diff
@@ -1,9 +1,7 @@
 import { readFileSync } from 'fs';
```

Comment:
__note:__ now this function is only used by `BasePathProxy`.

Reply:
Is it possible for the `BasePathProxy` to share connection setup logic with the new platform so we don't loose the shared setup? I think it's important that the default dev server use as close to a "production" setup as possible and keeping this and the new platform setup logic in sync doesn't feel like intuitive or likely to happen,

Reply:
>  I think it's important that the default dev server use as close to a "production" setup as possible and keeping this and the new platform setup logic in sync doesn't feel like intuitive or likely to happen,

Wholeheartedly agree, I just wanted to handle this in the follow-up, but let me check how hard it is to handle this here.

Reply:
So I started moving `BasePathProxy` to new platform and it's getting too big to fit into this PR, does it work for you if I create a follow-up right after this one @spalger ? The rest of the comments should be handled now.

Reply:
Will handle basepath proxy in a separate PR https://github.com/elastic/kibana/pull/19424